addons/point_of_sale/models/pos_session.py

```diff
@@ -1955,16 +1955,23 @@ def _get_pos_ui_pos_category(self, params):
         return self.env['pos.category'].search_read(**params['search_params'])
 
     def _loader_params_pos_combo(self):
-        return {'search_params': {'domain': [], 'fields': ['id', 'name', 'combo_line_ids']}}
+        return {'search_params': {'fields': ['id', 'name', 'combo_line_ids']}}
 
     def _get_pos_ui_pos_combo(self, params):
-        return self.env['pos.combo'].search_read(**params['search_params'])
+        return (self.env['product.product']
+                    .search(self.config_id._get_availlable_product_domain())
+                    .combo_ids
+                    .read(params['search_params']['fields']))
```

Comment:
loading the products is normally an expensive operation in pos and here, you will do it again. Maybe we can avoid doing that by looking in to the loaded_data? Derive the domain for po.combo using the loaded_data. you can find loaded_data from the context.

Reply:
> expensive operation in pos and here, you will do it again
i thought about this. I agree